### PR TITLE
Settings [cleanup]

### PR DESCRIPTION
**Pull request by ivangrek**
Continuation #8450, #8451

## Proposed changes

- Drop ValueOrDefault property
- Add IsUnset property

For non nullable
          Value = (value ?? Default), IsUnset = (value == null)
For nullable except "String", all values is significant
          Value = value, IsUnset = false

### After

image

image

## Test methodology 

- Manual

## Test environment(s) 

- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.28.0.windows.1
- Microsoft Windows NT 10.0.19041.0
- .NET Framework 4.8.4220.0
- DPI 96dpi (no scaling)

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by ivangrek** [comment]
@RussKie could you please do a review.

Cleaner as I think.
There are many more interesting things. 😄

**Review by RussKie — COMMENTED** [review]
I'd like to see tests associated with these changes, to ensure we don't regress in any way.
Users are never happy to lose settings.

File: GitCommands/Settings/Setting.cs [hunk 1]
```diff
@@ -57,70 +56,93 @@ public SettingOf(SettingsPath settingsSource, string name, T defaultValue = defa
...
                 {
                     if (Value.Equals(value))
                     {
                         return;
                     }
 
-                    SetValue(Name, value);
+                    if (Type.GetTypeCode(typeof(T)) == TypeCode.String)
+                    {
+                        SetValue(Name, (object)value ?? string.Empty);
```
[the remark this answers is not in the document]
Comment (ivangrek) [reply GitCommands/Settings/Setting.cs:90]: String as value type and we not save null to get IsUnset.

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you!

**Pull request merged**